src/services/outliningElementsCollector.ts

```diff
@@ -113,7 +111,9 @@ namespace ts.OutliningElementsCollector {
                 case SyntaxKind.SingleLineCommentTrivia:
                     // never fold region delimiters into single-line comment regions
                     const commentText = sourceText.slice(pos, end);
-                    if (isRegionDelimiter(commentText)) {
+                    const regionDelimiterRegExp = /^\s*\/\/\s*#(end)?region(?:\s+(.*))?(?:\r)?$/;

```

Comment:
Any reason why you have inlined this instead of returning from function.

Reply:
@amcasey thought it might read more clearly. https://github.com/microsoft/TypeScript/pull/31220#discussion_r281374038

Reply:
I would rather not inline it so we know there are two places the regexp needs to change if at all later.